src/process/terminalsource.rs

```diff
@@ -223,6 +232,81 @@ impl io::Write for ColorableTerminalLocked {
     }
 }
 
+impl TermLike for ColorableTerminal {
+    fn width(&self) -> u16 {
+        Term::stdout().size().1
```

Comment:
Bonus: `cargo` uses the [`CARGO_TERM_PROGRESS_WIDTH`](https://doc.rust-lang.org/cargo/reference/config.html#termprogresswidth) environment variable to control the terminal width being detected. I think it'd be interesting to mirror that here as well, maybe in a separate commit or PR.